core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/expression/comparison/CmpExpressionDsl.kt

```diff
@@ -0,0 +1,104 @@
+package com.github.inflab.spring.data.mongodb.core.aggregation.expression.comparison
+
+import com.github.inflab.spring.data.mongodb.core.aggregation.expression.AggregationExpressionDsl
+import com.github.inflab.spring.data.mongodb.core.annotation.AggregationMarker
+import com.github.inflab.spring.data.mongodb.core.extension.toDotPath
+import org.bson.Document
+import org.springframework.data.mongodb.core.aggregation.AggregationExpression
+import kotlin.reflect.KProperty
+
+/**
+ * A Kotlin DSL to configure `$cmp` [CmpExpressionDsl] using idiomatic Kotlin code.
+ *
+ * @author minwoo
+ * @since 1.0
+ * @see <a href="https://docs.mongodb.com/manual/reference/operator/aggregation/cmp">$cmp</a>
+ */
+@AggregationMarker
+class CmpExpressionDsl {
+    /**
+     * Represents all operands inside `$cmp` as a list.
+     *
+     * @property values The list that contains all operands.
+     */
+    @JvmInline
+    value class Operands(val values: MutableList<Any>)
+
+    /**
+     * Creates a [Operands] with the given [value].
+     *
+     * @param value The number.
+     */
+    fun of(value: Number) = Operands(mutableListOf(value))
+
+    /**
+     * Creates a [Operands] with the given [field].
+     *
+     * @param field The name of the field.
+     */
+    fun of(field: String) = Operands(mutableListOf("$$field"))
```

Comment:
we should provide a method to treat string as value.

Reply:
I have found that comparison operators can compare any BSON type.
so it would be better to create a method takes `Any`.